libs/execution_context/ebpf_maps.c

```diff
@@ -2922,10 +2922,16 @@ ebpf_map_get_next_key_and_value_batch(
 
         memcpy(key_and_value + output_length + key_size, next_value, value_size);
 
-        if (flags & EBPF_MAP_FIND_FLAG_DELETE) {
-            // If the caller requested deletion, delete the entry.
+        if ((flags & EBPF_MAP_FIND_FLAG_DELETE) && (previous_key != NULL)) {
+            // If the caller requested deletion, delete the previous entry.
             result = ebpf_map_metadata_tables[map->ebpf_map_definition.type].delete_entry(map, previous_key);
             if (result != EBPF_SUCCESS) {
+                EBPF_LOG_MESSAGE_UINT64_UINT64(
+                    EBPF_TRACELOG_LEVEL_ERROR,
+                    EBPF_TRACELOG_KEYWORD_MAP,
+                    "Failed to delete entry with key",
+                    *(uint64_t*)previous_key,
```

Comment:
What if the key is not 64bits in length? This will read invalid memory.

Reply:
Moved the trace to EBPF_LOG_MESSAGE_UINT64.